packages/react-scripts/scripts/init.js

```diff
@@ -110,13 +110,7 @@ module.exports = function(
     require.resolve(`${templateName}/package.json`, { paths: [appPath] })
   );
 
-  let templateJsonPath;
-  if (templateName) {
-    templateJsonPath = path.join(templatePath, 'template.json');
-  } else {
-    // TODO: Remove support for this in v4.
-    templateJsonPath = path.join(appPath, '.template.dependencies.json');
```

Comment:
This is legacy, and is not related to the templates feature we released.

It was never officially supported and wasn't documented, and was used for internal testing purposes previously.